src/asset-swapper/utils/market_operation_utils/index.ts

```diff
@@ -778,19 +782,57 @@ export class MarketOperationUtils {
         return { optimizerResult, wholeOrderPrice };
     }
 
+    /**
+     * Returns whether a swap is considered extremely small (for simpler routing to avoid over optimization which causes high revert rate)
+     *
+     * @param inputAmount : taker amount for sell, maker amount for buy
+     * @param inputAmountPerNativeWei : taker amount per native token in (wei) for sell, maker amount per native token in (wei) for buy
+     */
+    private isMicroSwap(inputAmount: BigNumber, inputAmountPerNativeWei: BigNumber): boolean {
+        // Only enable it on Optimism as it's experimental.
+        if (this.sampler.chainId !== ChainId.Optimism) {
+            return false;
+        }
+        const inputTokenValueInNativeWei = inputAmount.div(inputAmountPerNativeWei);
+
+        // If the value of input token amount is less than 1/100 of `this._nativeFeeTokenAmount`
+        // then it is considered as a micro swap.
+        // NOTE: gt is used because `takerAmountPerNativeWei` can be 0 when it's not available.
```

Comment:
nit: do you mean inputAmountPerNativeWei

Reply:
yes! fixed!